### ISSUE TITLE
Rename pgInt4 and pgInt8

### ISSUE THREAD

**Issue by bergmark** [tomjaguarpaw/haskell-opaleye#80]
Most `pg*` functions are named after the haskell type e.g. `pgTimeOfDay`, but there's `pgInt4` and `pgInt8` which have postgres names, and "The type names int2, int4, and int8 are extensions, which are also used by some other SQL database systems."

Haskell's `Int` isn't guaranteed to fit in an `int4`, only on 32bit systems? Here Int and Int64 are the same.

So I think it would make more sense to have `pgInt` and `pgInt64` instead since 64bit users can't safely map `Int` to `int4`.

**Comment by tomjaguarpaw** [comment]
Yes, this is a strange inconsistency.  I'm happy to take suggestions for a full revamping of the names of these things, preferably in conjunction with https://github.com/tomjaguarpaw/haskell-opaleye/issues/81.

**Comment by bergmark** [comment]
Hmm it's not quite as clear cut as i thought since there's also json and jsonb functions that are named after the pg type...

**Comment by k0001** [comment]
See the related #110 -- I'm sorry I wasn't aware of this issue before.

I agree with using the names `pgInt32` and `pgInt64` in any case.
